### DISCUSSION TITLE
Blocking element by has-text with any lower/upper/cammelcase

### DISCUSSION THREAD

**Discussion by desiregn** [uBlockOrigin/uAssets#13356] [Q&A] [answered]
### Prerequisites

- [x] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

ANY

### Describe the issue

Rule example:

`website.com##.div:has-text(chrome)`

Not blocking Chrome/CHROME/ChRome

So i Do

`website.com##.div:has-text(/chrome/Chrome/CHROME/ChRome/CHrome/i)`

etc

But it is inconvenient to fill out the regexp.
Better will be something like

`website.com##.div:has-text-nc(chrome)`

**-nc** - as no case (as example)

### uBlock Origin version

1.42.4

### Browser name and version

Firefox 99.0 (64-bit)

### Settings

?

**Comment by mapx-** [comment, marked as answer]
If you want to hide all those "chrome" flavours just use

`website.com##.div:has-text(/chrome/i)`

**Reply by krystian3w** [reply]
Maybe words have 0-width characters between letters:

> `C&ZeroWidthSpace;hro&shy;me` - IMO github delete both pasted as html enity

Then regex need minimum:

`/c.hro.me/i` because who remember use "\u00AD" or "\u200B".

**Comment by krystian3w** [comment]
If you add detect enters (https://github.com/MajkiIT/polish-ads-filter/issues/20157#issuecomment-902916488) I will use `:has-text-nc` instead late time to write RegExp:

```adb
/word(\r|\n)+Word-where-enters-simulate-space/
```

```adb
/word\R+Word-where-enters-simulate-space/
```
